src/proxy/http2/Http2ConnectionState.cc

```diff
@@ -692,6 +704,11 @@ Http2ConnectionState::rcv_rst_stream_frame(const Http2Frame &frame)
     Http2StreamDebug(this->session, stream_id, "Parsed RST_STREAM frame: Error Code: %u", rst_stream.error_code);
     stream->set_rx_error_code({ProxyErrorClass::TXN, static_cast<uint32_t>(rst_stream.error_code)});
     stream->initiating_close();
+    // Delete the stream immediately to free up the stream count before the next frame is processed.
+    // This prevents the race condition where HEADERS frames are incorrectly refused due to the
+    // stream count not being decremented yet. The destructor will handle the case where the stream
+    // has already been removed from the stream list.
+    this->delete_stream(stream);
```

Comment:
Shouldn't we call `THREAD_FREE(stream, http2StreamAllocator, this_ethread());` instead of calling these two, `initiating_close` and `delete_stream`?